domain/model/mesh_results.py

```diff
@@ -1,21 +1,57 @@
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 from datetime import datetime, timezone
-from typing import List, Optional
+from typing import Dict, List, Optional, Tuple
 
+from domain.geo import Coordinates
+from domain.metric_type import MetricType
 from domain.types import AgentID, MetricValue
 
 
+@dataclass
+class Agent:
+    id: AgentID = AgentID()
+    ip: str = ""
+    name: str = ""
+    alias: str = ""
+    coords: Coordinates = Coordinates()
+
+
+class Agents:
+    def __init__(self) -> None:
+        self._agents: Dict[AgentID, Agent] = {}
+
+    def get_by_id(self, id: AgentID) -> Agent:
+        if id in self._agents:
```

Comment:
A simpler version:
```
return self._agents.get(id, Agent())
```
BTW, why not return `None` when agent with the `id` does not exist? It makes detecting failures and bugs much easier (IMHO).

Reply:
NullObject pattern currently seems more suitable here